src_erl/Communication_Layer/http_Nerlserver/src/source/sourceStatem.erl

```diff
@@ -132,11 +136,12 @@ castingData(cast, {leftOvers,Tail}, State = #source_statem_state{msgCounter = Co
 %%  io:format("received leftovers- ~p~n",[Tail]),
   {next_state, idle, State#source_statem_state{msgCounter = Counter+1,csvList = Tail}};
 
-castingData(cast, {finishedCasting,CounterReceived,ListOfSamples}, State = #source_statem_state{myName = MyName, msgCounter = Counter, portMap = PortMap}) ->
-  io:format("source finished casting- ~n",[]),
-  {RouterHost,RouterPort} = maps:get(mainServer,PortMap),
+castingData(cast, {finishedCasting,CounterReceived,ListOfSamples}, State = #source_statem_state{myName = MyName, msgCounter = Counter, nerlnetGraph = NerlnetGraph}) ->
+   io:format("source finished casting~n"),
```

Comment:
Should be more informative: Source + ID finished casting 

Try to envelope prints with debug or info flags